pcp/PCPDynamicMeter.c

```diff
@@ -427,22 +427,22 @@ void PCPDynamicMeter_display(PCPDynamicMeter* this, ATTR_UNUSED const Meter* met
             break;
          case PM_TYPE_32:
             len = conv.dimSpace ?
-               Meter_humanUnit(buffer, atom.l, sizeof(buffer)) :
+               Meter_humanUnit(buffer, (double)atom.l, sizeof(buffer)) :
                xSnprintf(buffer, sizeof(buffer), "%d", atom.l);
             break;
          case PM_TYPE_U32:
             len = conv.dimSpace ?
-               Meter_humanUnit(buffer, atom.ul, sizeof(buffer)) :
+               Meter_humanUnit(buffer, (double)atom.ul, sizeof(buffer)) :
                xSnprintf(buffer, sizeof(buffer), "%u", atom.ul);
             break;
          case PM_TYPE_64:
             len = conv.dimSpace ?
-               Meter_humanUnit(buffer, atom.ll, sizeof(buffer)) :
+               Meter_humanUnit(buffer, (double)atom.ll, sizeof(buffer)) :
                xSnprintf(buffer, sizeof(buffer), "%lld", (long long) atom.ll);
             break;
          case PM_TYPE_U64:
             len = conv.dimSpace ?
-               Meter_humanUnit(buffer, atom.ull, sizeof(buffer)) :
+               Meter_humanUnit(buffer, (double)atom.ull, sizeof(buffer)) :
```

Comment:
Does htop's coding style prefer implicit or explicit casts from integers to floating points?

Reply:
Implicit is sufficient … If no explicit cast is necessary nor does improve readability, it's preferred to go without. Reduce potential noise when reading the code.

Reply:
In this special case though (pmAtomValue), I'd prefer an explicit cast - otherwise it may look to the casual observer that atom.d should have been used here.